pkg/common/cns-lib/volume/util.go

```diff
@@ -41,36 +42,92 @@ func validateManager(ctx context.Context, m *defaultManager) error {
 
 // IsDiskAttached checks if the volume is attached to the VM.
 // If the volume is attached to the VM, return disk uuid of the volume, else return empty string
-func IsDiskAttached(ctx context.Context, vm *cnsvsphere.VirtualMachine, volumeID string) (string, error) {
+func IsDiskAttached(ctx context.Context, vm *cnsvsphere.VirtualMachine, volumeID string, checkNVMeController bool) (string, error) {
 	log := logger.GetLogger(ctx)
 	// Verify if the volume id is on the VM backing virtual disk devices
 	vmDevices, err := vm.Device(ctx)
 	if err != nil {
 		log.Errorf("failed to get devices from vm: %s", vm.InventoryPath)
 		return "", err
 	}
+	nvmeControllerKeyToNameMap := make(map[int32]string)
+	for _, device := range vmDevices {
+		if vmDevices.TypeName(device) == "VirtualNVMEController" {
+			var controllerName string
+			if device.GetVirtualDevice().DeviceInfo.GetDescription() != nil {
+				controllerName = device.GetVirtualDevice().DeviceInfo.GetDescription().Label
+			}
+			nvmeControllerKeyToNameMap[device.GetVirtualDevice().Key] = controllerName
+		}
+	}
+
 	for _, device := range vmDevices {
 		if vmDevices.TypeName(device) == "VirtualDisk" {
-			if virtualDisk, ok := device.(*vimtypes.VirtualDisk); ok {
+			if virtualDisk, ok := device.(*types.VirtualDisk); ok {
 				if virtualDisk.VDiskId != nil && virtualDisk.VDiskId.Id == volumeID {
 					virtualDevice := device.GetVirtualDevice()
-					if backing, ok := virtualDevice.Backing.(*vimtypes.VirtualDiskFlatVer2BackingInfo); ok {
+					if value, ok := nvmeControllerKeyToNameMap[virtualDevice.ControllerKey]; ok {
+						log.Info("Found that the disk in contention is attached to NVMe controller")
```

Comment:
Instead of this static log message, how about this:

`Found disk %s is attached to NVMe controller on VM %s.`

And could you please move the log to debug?

Reply:
Done